.github/workflows/maven.yaml

```diff
@@ -82,3 +89,56 @@ jobs:
         with:
           name: container-logs
           path: "**/container-logs/**/*.log"
+
+  older_kafka_matrix:
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      matrix:
+        version: [3.8.0]
+    steps:
+      - name: 'Check out repository'
+        uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+          fetch-depth: 0
+
+      - name: 'Set up Java'
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'zulu'
+          cache: maven
+
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: 'Install junit extension'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: mvn -B install -DskipTests=true
+
+      - name: 'Run select junit tests (Kafka in-VM)'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # Note - the extension won't necessarily compile against the earlier versions of Kafka.
+        # We want to run the tests against the code that was compiled in the install step, just with
+        # a different version of Kafka on the classpath.
+        run: mvn -Dmaven.main.skip=true -Dmaven.javadoc.skip=true -pl impl,junit5-extension -B test -Dtest=KafkaClusterTest,ParameterExtensionTest -Dkafka.version=${{ matrix.version }}
```

Comment:
Would it be easier if we added a `crossVersion` (better name required) profile to control the set of tests which are covered?

Reply:
Yes, I think that would be an improvement, but I think it can wait.